File: coco/src/main/java/coco/ep/i2/finalized/Power.java
package coco.ep.i2.finalized;

import coco.ep.i1.finalized.Exp;

public class Power extends Exp implements Factory, coco.ep.i2.Power<Exp> {

    protected Exp left;
    protected Exp right;

    public Power(Exp _left, Exp _right) {
        this.left = _left;
        this.right = _right;
    }

    public Exp getLeft() {
        return this.left;
    }
    public Exp getRight() {
        return this.right;
    }
}

File: coco_csharp/CoCo/EP/I2/Finalized/Power.cs
namespace EP.I2.Finalized {
    public class Power : EP.I1.Finalized.Exp, EP.I2.Power<EP.I1.Finalized.Exp>, Factory {
        private EP.Exp<EP.I1.Finalized.Exp> left;
        private EP.Exp<EP.I1.Finalized.Exp> right;

        public Power(EP.Exp<EP.I1.Finalized.Exp> _left, EP.Exp<EP.I1.Finalized.Exp> _right) {
            this.left = _left;
            this.right = _right;
        }

        public EP.Exp<EP.I1.Finalized.Exp> getLeft() { return this.left; }

        public EP.Exp<EP.I1.Finalized.Exp> getRight() { return this.right; }
    }
}